[12:13:41 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: Route.get() requires a callback function but got a [object Undefined]\n    at Route.<computed> [as get] (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\node_modules\\express\\lib\\router\\route.js:216:15)\n    at proto.<computed> [as get] (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\node_modules\\express\\lib\\router\\index.js:521:19)\n    at Object.<anonymous> (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\routes\\admin.js:252:8)\n    at Module._compile (node:internal/modules/cjs/loader:1554:14)\n    at Object..js (node:internal/modules/cjs/loader:1706:10)\n    at Module.load (node:internal/modules/cjs/loader:1289:32)\n    at Function._load (node:internal/modules/cjs/loader:1108:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)\n    at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)\n    at Module.require (node:internal/modules/cjs/loader:1311:12)","name":"Error","message":"Route.get() requires a callback function but got a [object Undefined]"}
[12:42:05 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: Cannot find module './controllers/admin/contact'\nRequire stack:\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\routes\\admin.js\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js\n    at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)\n    at Function._load (node:internal/modules/cjs/loader:1055:27)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)\n    at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)\n    at Module.require (node:internal/modules/cjs/loader:1311:12)\n    at require (node:internal/modules/helpers:136:16)\n    at Object.<anonymous> (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\routes\\admin.js:14:27)\n    at Module._compile (node:internal/modules/cjs/loader:1554:14)\n    at Object..js (node:internal/modules/cjs/loader:1706:10)\n    at Module.load (node:internal/modules/cjs/loader:1289:32)","name":"Error","message":"Cannot find module './controllers/admin/contact'\nRequire stack:\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\routes\\admin.js\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js"}
[12:42:26 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: Cannot find module './controllers/admin/contact'\nRequire stack:\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\routes\\admin.js\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js\n    at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)\n    at Function._load (node:internal/modules/cjs/loader:1055:27)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)\n    at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)\n    at Module.require (node:internal/modules/cjs/loader:1311:12)\n    at require (node:internal/modules/helpers:136:16)\n    at Object.<anonymous> (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\routes\\admin.js:14:27)\n    at Module._compile (node:internal/modules/cjs/loader:1554:14)\n    at Object..js (node:internal/modules/cjs/loader:1706:10)\n    at Module.load (node:internal/modules/cjs/loader:1289:32)","name":"Error","message":"Cannot find module './controllers/admin/contact'\nRequire stack:\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\routes\\admin.js\n- C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js"}
[12:50:33 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"TypeError: EmailService is not a constructor\n    at Object.<anonymous> (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\controllers\\admin\\contact.js:3:22)\n    at Module._compile (node:internal/modules/cjs/loader:1554:14)\n    at Object..js (node:internal/modules/cjs/loader:1706:10)\n    at Module.load (node:internal/modules/cjs/loader:1289:32)\n    at Function._load (node:internal/modules/cjs/loader:1108:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)\n    at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)\n    at Module.require (node:internal/modules/cjs/loader:1311:12)\n    at require (node:internal/modules/helpers:136:16)\n    at Object.<anonymous> (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\controllers\\admin\\index.js:8:27)","name":"TypeError","message":"EmailService is not a constructor"}
[20:24:10 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: listen EADDRINUSE: address already in use :::5000\n    at Server.setupListenHandle [as _listen2] (node:net:1937:16)\n    at listenInCluster (node:net:1994:12)\n    at Server.listen (node:net:2099:7)\n    at Function.listen (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\node_modules\\express\\lib\\application.js:635:24)\n    at startServer (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js:363:13)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","name":"Error","message":"listen EADDRINUSE: address already in use :::5000"}
[21:51:24 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: listen EADDRINUSE: address already in use :::5000\n    at Server.setupListenHandle [as _listen2] (node:net:1937:16)\n    at listenInCluster (node:net:1994:12)\n    at Server.listen (node:net:2099:7)\n    at Function.listen (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\node_modules\\express\\lib\\application.js:635:24)\n    at startServer (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js:363:13)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","name":"Error","message":"listen EADDRINUSE: address already in use :::5000"}
[21:53:41 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: listen EADDRINUSE: address already in use :::5000\n    at Server.setupListenHandle [as _listen2] (node:net:1937:16)\n    at listenInCluster (node:net:1994:12)\n    at Server.listen (node:net:2099:7)\n    at Function.listen (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\node_modules\\express\\lib\\application.js:635:24)\n    at startServer (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js:363:13)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","name":"Error","message":"listen EADDRINUSE: address already in use :::5000"}
[21:54:43 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: listen EADDRINUSE: address already in use :::5000\n    at Server.setupListenHandle [as _listen2] (node:net:1937:16)\n    at listenInCluster (node:net:1994:12)\n    at Server.listen (node:net:2099:7)\n    at Function.listen (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\node_modules\\express\\lib\\application.js:635:24)\n    at startServer (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js:363:13)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","name":"Error","message":"listen EADDRINUSE: address already in use :::5000"}
[21:55:14 23/07/2025] [ERROR] Uncaught Exception | Meta: {"fatal":true,"stack":"Error: listen EADDRINUSE: address already in use :::5000\n    at Server.setupListenHandle [as _listen2] (node:net:1937:16)\n    at listenInCluster (node:net:1994:12)\n    at Server.listen (node:net:2099:7)\n    at Function.listen (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\node_modules\\express\\lib\\application.js:635:24)\n    at startServer (C:\\Users\\Cat Thach\\Desktop\\monhoc\\T5_cdbackend\\lab_node\\app.js:363:13)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","name":"Error","message":"listen EADDRINUSE: address already in use :::5000"}
